prow/entrypoint/run.go

```diff
@@ -206,3 +211,20 @@ func gracefullyTerminate(command *exec.Cmd, done <-chan error, gracePeriod time.
 		}
 	}
 }
+
+// gitHeadTimestamp returns the timestamp of the HEAD commit as seconds from the
+// UNIX epoch. If unable to read the timestamp for any reason (such as missing
+// the git, or not using a git repo), returns 0 and an error.
+func gitHeadTimestamp() (int, error) {
+	gitOutput, err := exec.Command("git", "show", "-s", "--format=format:%ct", "HEAD").Output()
```

Comment:
Why do we need this on the entrypoint? This also assumes the test image will have `git` which is not the case today

Reply:
this is basically just following https://reproducible-builds.org/docs/source-date-epoch/#git-repository - we want to set `SOURCE_DATE_EPOCH` to the timestamp of the last modification.

I hadn't thought about the test image not having `git`. In kubernetes we would pretty much require it to get a valid version string, but I guess that's not the case everywhere. We could instead save the git timestamp (and maybe other info?) into a file somewhere? Not sure where though.

Reply:
(also, this fails silently - if `git` isn't available, we'll just not set `SOURCE_DATE_EPOCH`, which is the existing behavior.)

Reply:
Yeah pod-utils doesn't want to provide any restriction on the image used. Can we thread through the data using file passing in shared volumes? You can see how we do this for other data in decorate.go

Reply:
I guess you are making a reasonable argument that this doesn't belong in `entrypoint`.

Alternatives I see:
a) Add it to the bootstrap image, which would cover the majority of kubernetes' needs (and would be easy to implement). Anyone not using a bootstrap-derived image would need to implement this themselves.
b) Make clonerefs save out metadata to a separate file (either just the timestamp, or maybe additionally other git info like `git describe`?) We could then either plumb this file through or pass it via an environment variable or both. I'm not really sure what would be best. This would be quite a bit more work, and I'm not sure how I'd even test it, since we don't really have any integration tests for the pod-utils components.

At this point I'm leaning towards a), since it unblocks our migration to pod-utils. If anyone wants a more general implementation, they're welcome to work on it.

Reply:
I think `b)` would be a good approach for the future -- I'm thinking something like adding a flag to point to the location where to write the data, then throwing down a new volume shared between clonerefs and entrypoint, and similarly plumbing through a flag to entrypoint to consume it.